SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -707,6 +747,59 @@ void CMeshSolver::SetBoundaryDisplacements(CGeometry *geometry, CNumerics *numer
 
 }
 
+void CMeshSolver::SetBoundaryVelocities(CGeometry *geometry, CNumerics *numerics, CConfig *config){
+
+  unsigned short iMarker;
+
+  /*--- Impose zero displacements of all non-moving surfaces (also at nodes in multiple moving/non-moving boundaries). ---*/
+  /*--- Exceptions: symmetry plane, the receive boundaries and periodic boundaries should get a different treatment. ---*/
+  for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
```

Comment:
Maybe we can give 2 modes to SetBoundaryDisplacement (displacement and velocity) to avoid duplicating the logic of what to do with each marker (which is not the same anymore btw).

Reply:
Yes I think this would be better. For the velocity boundary, I just needed to set the boundary constraints for all the markers in the same way as that used for the displacement boundary but then switch BC_Deforming to BC_Velocity. So I've created a boolean flag now to switch between the two within SetBoundaryDisplacement.